pkg/types/llm/config.go

```diff
@@ -31,37 +31,37 @@ type Config struct {
 	Aliases              map[string]string  // Aliases maps short model names to full model names
 	Retry                RetryConfig        // Retry configuration for API calls
 
-	// Provider-specific configurations
-	OpenAI   *OpenAIConfig           `mapstructure:"openai"`   // OpenAI-specific configuration including compatible providers
-	SubAgent *SubAgentConfigSettings `mapstructure:"subagent"` // SubAgent configuration for different models/providers
+	Profile  string                   `mapstructure:"profile"`
+	Profiles map[string]ProfileConfig `mapstructure:"profiles"`
+
+	OpenAI   *OpenAIConfig           `mapstructure:"openai"`
+	SubAgent *SubAgentConfigSettings `mapstructure:"subagent"`
 }
 
-// OpenAIConfig holds OpenAI-specific configuration including support for compatible APIs
 type OpenAIConfig struct {
-	Preset       string                  `mapstructure:"preset"`          // Built-in preset for popular providers (e.g., "xai")
```

Comment:
the comments on the struct fields holdn't be removed - they need to be reverted back